sellercenter-attributes/src/main/java/com/gfg/sellercenter/attributes/service/AttributeService.java

```diff
@@ -0,0 +1,72 @@
+package com.gfg.sellercenter.attributes.service;
+
+import com.gfg.sellercenter.attributes.entity.AttributeOption;
+import com.gfg.sellercenter.attributes.entity.AttributeWithOption;
+import com.gfg.sellercenter.attributes.entity.ProductWithAttributeInformation;
+import com.gfg.sellercenter.attributes.reader.HttpReader;
+
+import lombok.AllArgsConstructor;
+
+import org.json.JSONArray;
+import org.json.JSONObject;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+@AllArgsConstructor
+public class AttributeService {
+
+    private final HttpReader httpReader;
+
+    public Map<Integer, ProductWithAttributeInformation>
+            getSerialNumberRequiredAttributesByProductIds(List<Integer> productIds)
+                    throws Exception {
+        productIds = productIds.stream().distinct().collect(Collectors.toList());
+
+        return createAttributeEntity(
+                this.httpReader.getSerialNumberRequiredAttributesByProductIds(productIds));
+    }
+
+    private Map<Integer, ProductWithAttributeInformation> createAttributeEntity(
+            JSONArray responseArray) {
+        Map<Integer, ProductWithAttributeInformation> categoryToProductMap = new HashMap<>();
+
+        for (Object response : responseArray) {
```

Comment:
will it work with? 
```suggestion
        for (JSONObject response : responseArray) {
```

Reply:
no, IDE tells me that the code is invalid it requires a type of Object but got JSONObject same for if i would use lambda.